File: code/data/enum.c
#include <stdio.h>

typedef enum {
    Seg,
    Ter,
    Qua,
    Qui,
    Sex,
    Sab,
    Dom,
} Dias;

int main (void) {
    Dias hoje = Sex;
    printf(">>> %d\n", hoje);
    hoje = 1;
    printf(">>> %d\n", hoje);
    return 0;
}

File: code/data/enum.hs
data Dias = Seg | Ter | Qua | Qui | Sex | Sab | Dom

paraNome :: Dias -> String
paraNome d =
  case d of
    Sab -> "Sábado"
    Dom -> "Domingo"
    --otherwise -> "Dia de Trabalho"
    
hoje = Seg

main = print (paraNome hoje)
